src/RA_Dlg_AchEditor.cpp

```diff
@@ -246,25 +239,43 @@ LRESULT CALLBACK EditProc(HWND hwnd, UINT nMsg, WPARAM wParam, LPARAM lParam) no
             lvDispinfo.item.iSubItem = nSelSubItem;
             lvDispinfo.item.pszText = nullptr;
 
-            TCHAR sEditText[12];
-            GetWindowText(hwnd, sEditText, 12);
-            lvDispinfo.item.pszText = sEditText;
-            lvDispinfo.item.cchTextMax = lstrlen(sEditText);
+            std::array<char, 12> sEditText{};
+            GetWindowTextA(hwnd, sEditText.data(), 12);
+            lvDispinfo.item.pszText = sEditText.data();
+            lvDispinfo.item.cchTextMax = gsl::narrow_cast<int>(std::strlen(sEditText.data()));
 
             HWND hList = GetDlgItem(g_AchievementEditorDialog.GetHWND(), IDC_RA_LBX_CONDITIONS);
 
             // the LV ID and the LVs Parent window's HWND
             SendMessage(GetParent(hList), WM_NOTIFY, static_cast<WPARAM>(IDC_RA_LBX_CONDITIONS),
                         reinterpret_cast<LPARAM>(&lvDispinfo)); // ##reinterpret? ##SD
 
-            // Extra validation?
-
-            if (lstrcmp(sEditText, TEXT("Value")))
+            if (g_AchievementEditorDialog.LbxDataAt(lvDispinfo.item.iItem, CondSubItems::Type_Tgt).compare("Value") == 0)
             {
+                constexpr std::array<unsigned char, 25> valid_chars{"0123456789ABCDEFabcdefXx"};
                 // Remove the associated 'size' entry
-                strcpy_s(g_AchievementEditorDialog.LbxDataAt(lvDispinfo.item.iItem,
-                                                             ra::itoe<CondSubItems>(lvDispinfo.item.iSubItem + 1)),
-                         32, "");
+                for (const unsigned char& c : sEditText)
+                {
+                    if (c == unsigned char())
+                        break;
+                    
+#pragma warning(suppress: 26447) // f.6, inline suppression not working here but works for message box
+                     if (!std::any_of(valid_chars.begin(), valid_chars.end(),
+                                     [&c](unsigned char valid) noexcept { return c == valid; }))
+                    {
+                        GSL_SUPPRESS_F6
+                        ra::ui::viewmodels::MessageBoxViewModel::ShowWarningMessage(
+                            L"Only values that can be represented as hex or dec are allowed.");
+                        DestroyWindow(hwnd);
+                        return 0;
+                    }
+                }
+                if ((lvDispinfo.item.iItem >= 0) && (lvDispinfo.item.iSubItem >= 1))
+                {
+                    g_AchievementEditorDialog
+                        .LbxDataAt(lvDispinfo.item.iItem, ra::itoe<CondSubItems>(lvDispinfo.item.iSubItem))
+                        .clear();
+                }
```

Comment:
I'm just typing in a garbage string to see if an exception is thrown. 
(not my changes) There's some weird achievements in local it warns me that they exist on the server. I think that's a bug. 

I couldn't reproduce the exception.
How can I get that LVN_ENDLABELEDIT exception  to happen? The validation was made to prevent crashes.

Like type in "$@!#!@$Jfdjka;" and it'll give you the warning message. I'm not trying to be sarcastic but isn't "bad" valid? It would be written as "0xbad", unless letters aren't allowed which I thought they were in hex.

What would you suggest? Just get rid of it?

Reply:
"bad" was sufficient to throw an exception. it was the std::stoul code noted below. Did you try to reproduce it before changing back to `strtoul`? 

further testing indicates "bad" is valid (and converted to 0xbad) if "show decimal values" is unchecked.